chapters/functions.tex

```diff
@@ -2372,6 +2373,8 @@ \subsection{Annotations for External Functions}\label{annotations-for-external-l
   In case calls to several external functions are generated in the same translation unit, the \lstinline!Include! annotations of the different functions must not define the same function -- except when relying on the deprecated behavior.
 
   The included code should be valid C89 code.
+  If the \lstinline[language=grammar]!external-function-call! contains any `size`-expression, the tool is responsible for ensuring that a C-header defining \lstinline[language=C]!size_t! is included before the \lstinline!"insertedCode"!.
+  The \lstinline!"insertedCode"! (following the inclusion of a header for \lstinline[language=C]!size_t! when this would be a tool responsibility) shall be possible to compile separately.
```

Comment:
I agree with the intention of this, but the formulation is now quite complicated. It has conditionals and parts in parenthesis.
Could we simplify it to something like:
```suggestion
  This gives a code fragment \lstinline!"insertedCode"! conditionally preceded by a header for \lstinline[language=C]!size_t!.
  This code fragment shall be possible to compile separately.
```

Reply:
@henrikt-ma ?

Reply:
Sure, it would be nice to simplify.  However, I don't find the suggestion clear enough.  Let me try something else:
```suggestion
  With at most the tool-supplied inclusion of a header for \lstinline[language=C]!size_t! prepended, the \lstinline!"insertedCode"! shall be possible to compile separately.
```

Reply:
Now I also see the intention of your suggestion, namely _a code fragment consisting of…_.  That could work too, but I wouldn't call it a _code fragment_ considering it's something as complete as a translation unit.

Reply:
I prefer Hans's variant with two sentences with a small change to the second sentence.
```suggestion
This gives a code fragment \lstinline!"insertedCode"! conditionally preceded by a header for \lstinline[language=C]!size_t!.
It must be possible to compile this code fragment separately.
```